modules/rpc/handler.go

```diff
@@ -88,3 +90,27 @@ func authorize(ctx context.Context, host service.Host) (context.Context, error)
 	}
 	return ctx, nil
 }
+
+type panicInboundMiddleware struct{}
+
+func (p panicInboundMiddleware) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter, handler transport.UnaryHandler) error {
+	defer panicRecovery(ctx)
+	return handler.Handle(ctx, req, resw)
+}
+
+type panicOnewayInboundMiddleware struct{}
+
+func (p panicOnewayInboundMiddleware) HandleOneway(ctx context.Context, req *transport.Request, handler transport.OnewayHandler) error {
+	defer panicRecovery(ctx)
+	return handler.HandleOneway(ctx, req)
+}
+
+func panicRecovery(ctx context.Context) {
+	if err := recover(); err != nil {
+		stats.RPCPanicCounter.Inc(1)
+		fx.Logger(ctx).Error("Panic recovered serving request", "error", err)
+		// rethrow panic back to yarpc
+		// before https://github.com/yarpc/yarpc-go/issues/734 fixed, throw a generic error.
+		panic(panicRespnose)
```

Comment:
I think we should return general error message to client (since client can be something out of DC) so we don't leak something like: server use golang, server has nil pointer etc. Meaningful info can be present through introspector and sentry to service owners .